### DISCUSSION TITLE
Unable to Push file to S3 bucket from ECS Container

### DISCUSSION THREAD

**Discussion by sundar-tonik** [aws/aws-sdk-java#2742] [Q&A]
Unable to Push file to S3 bucket from ECS Container.
But the same code is used to push file to S3 from local container with the same policies maintained in both the places.
Unable to receive any exception from the ECS Container logs, execution moves to finally block.
But the same code executes successfully from local container.

` try {
            
            this.filePath = (String)context.getProperty("fileNamenPath");
            final File f = new File(this.filePath);
            final AmazonS3ClientBuilder as3c = AmazonS3ClientBuilder.standard();
            this.localkeyName = ((String)context.getProperty("KeyName")).concat((String)context.getProperty("fileName"));
            DefaultAWSCredentialsProviderChain.getInstance());
            DefaultAWSCredentialsProviderChain.getInstance().getCredentials());
            final AmazonS3 s3Client = (AmazonS3)new AmazonS3Client((AWSCredentialsProvider)DefaultAWSCredentialsProviderChain.getInstance());
            this.tm = TransferManagerBuilder.standard().withS3Client(s3Client).build();
            final Upload upload = this.tm.upload("BucketName", this.localkeyName, f);
            upload.waitForException();
        }
        catch (AmazonServiceException e) {
            e.printStackTrace();
        }
        catch (SdkClientException e2) {
            e2.printStackTrace();
        }
        catch (InterruptedException ie) {
            ie.printStackTrace();
        }
        catch (Exception e3) {
            e3.printStackTrace();
        }
        finally {
            System.out.println("Inside Custom Mediator -- inside Finally");
            this.tm.shutdownNow();
        }`

**Comment by debora-ito** [comment]
@sundarramangit It's hard to give suggestions when there's no error in the logs... 
Have you checked the correct bucket in the correct region? 
Can you enable the verbose wirelogs? Make sure to redact any sensitive information the logs might contain.
https://docs.aws.amazon.com/sdk-for-java/v1/developer-guide/java-dg-logging.html#sdk-net-logging-verbose
